CMakeLists.txt

```diff
@@ -27,60 +41,74 @@ set(rqt_image_view_UIS
 
 set(rqt_image_view_INCLUDE_DIRECTORIES
   include
-  "${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_INCLUDE_DESTINATION}"
+  ${rclcpp_INCLUDE_DIRS}
+  ${qt_gui_cpp_INCLUDE_DIRS}
+  ${rqt_gui_cpp_INCLUDE_DIRS}
+  ${image_transport_INCLUDE_DIRS}
+  ${sensor_msgs_INCLUDE_DIRS}
+  ${geometry_msgs_INCLUDE_DIRS}
+  ${cv_bridge_INCLUDE_DIRS}
+  ${Qt5Widgets_INCLUDE_DIRS}
 )
-if(NOT EXISTS "${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_INCLUDE_DESTINATION}")
-  file(MAKE_DIRECTORY "${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_INCLUDE_DESTINATION}")
-endif()
 
-catkin_package(
-  INCLUDE_DIRS ${rqt_image_view_INCLUDE_DIRECTORIES}
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS rqt_gui rqt_gui_cpp image_transport sensor_msgs cv_bridge geometry_msgs
+qt5_wrap_cpp(rqt_image_view_MOCS ${rqt_image_view_HDRS})
+
+qt5_wrap_ui(rqt_image_view_UIS_H ${rqt_image_view_UIS})
+
+include_directories(
+  ${rqt_image_view_INCLUDE_DIRECTORIES}
 )
-catkin_python_setup()
 
-if("${qt_gui_cpp_USE_QT_MAJOR_VERSION} " STREQUAL "5 ")
-  qt5_wrap_cpp(rqt_image_view_MOCS ${rqt_image_view_HDRS})
-else()
-  qt4_wrap_cpp(rqt_image_view_MOCS ${rqt_image_view_HDRS})
-endif()
-# ensure generated header files are being created in the devel space
-set(_cmake_current_binary_dir "${CMAKE_CURRENT_BINARY_DIR}")
-set(CMAKE_CURRENT_BINARY_DIR "${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_INCLUDE_DESTINATION}")
-if("${qt_gui_cpp_USE_QT_MAJOR_VERSION} " STREQUAL "5 ")
-  qt5_wrap_ui(rqt_image_view_UIS_H ${rqt_image_view_UIS})
-else()
-  qt4_wrap_ui(rqt_image_view_UIS_H ${rqt_image_view_UIS})
-endif()
-set(CMAKE_CURRENT_BINARY_DIR "${_cmake_current_binary_dir}")
-
-include_directories(${rqt_image_view_INCLUDE_DIRECTORIES} ${catkin_INCLUDE_DIRS})
-add_library(${PROJECT_NAME} ${rqt_image_view_SRCS} ${rqt_image_view_MOCS} ${rqt_image_view_UIS_H})
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
-if("${qt_gui_cpp_USE_QT_MAJOR_VERSION} " STREQUAL "5 ")
-  target_link_libraries(${PROJECT_NAME} Qt5::Widgets)
-else()
-  target_link_libraries(${PROJECT_NAME} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
-endif()
+add_library(${PROJECT_NAME} SHARED
+  ${rqt_image_view_SRCS}
+  ${rqt_image_view_MOCS}
+  ${rqt_image_view_UIS_H}
+)
 
-find_package(class_loader)
-class_loader_hide_library_symbols(${PROJECT_NAME})
+target_link_libraries(
+  ${PROJECT_NAME}
+  ${rclcpp_LIBRARIES}
+  ${qt_gui_cpp_LIBRARIES}
+  ${rqt_gui_cpp_LIBRARIES}
+  ${image_transport_LIBRARIES}
+  ${sensor_msgs_LIBRARIES}
+  ${geometry_msgs_LIBRARIES}
+  ${cv_bridge_LIBRARIES}
+  Qt5::Widgets
+)
 
-install(FILES plugin.xml
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+install(
+  TARGETS ${PROJECT_NAME}
+  EXPORT ${PROJECT_NAME}
+  ARCHIVE DESTINATION lib
```

Comment:
Package specific project names

Reply:
Also several lines below

Reply:
This is taken from here: https://github.com/ros-visualization/qt_gui_core/blob/00eee66c53fb105b8b55f57b2f01f85902fb4b04/qt_gui_cpp/src/qt_gui_cpp/CMakeLists.txt#L43

and here: https://github.com/ros2/rviz/blob/31b35f99b870421f0780480276a0a71e67aaa7e0/rviz_default_plugins/CMakeLists.txt#L257

I'm not sure what you are suggesting I change. Can you suggest a commit?